server/journeys/base/controller.ts

```diff
@@ -52,4 +56,12 @@ export class BaseJourneyController {
         checked: Array.isArray(value) ? value.includes(refData.code) : refData.code === value,
       }))
       .sort((a, b) => sortAscending(a.text, b.text))
+
+  getRecordedByFieldsFromJwt = (token: string) => {
+    const jwt = jwtDecode(token) as { user_name: string; name: string }
+    return {
+      recordedBy: jwt.user_name,
+      recordedByDisplayName: jwt.name,
+    }
+  }
```

Comment:
these are decoded and available from `res.locals.user.username` and `res.locals.user.displayName`

Reply:
Excellent, I wasn't aware. This simplifies a lot, thank you